### PR TITLE
#571 Improve performance on parameters reading and injection

### PR DESCRIPTION
**Pull request by rosiereflo**

### CODE REVIEW

**Comment by rosiereflo** [comment]
Decrease parameters handle time by
- Avoiding to search multiple times in a map for the parameter object
- Avoiding reading same file multiple times

File: dynawo/sources/Common/Dictionaries/DYNLog_en_GB.dic [hunk 1]
```diff
@@ -181,6 +181,7 @@ SolverVarDisplayLimit         =             display limited to %1% variables
 SolverNbZVar                  =             number of discrete variables : %1%
 // --> DYNSolverSIM
 SolverSIMInitOK               =             initialization of SIM solver : ok
+SolverSIMInitGuessOK          =             initial guess of KINSOL is accurate, following solver iterations will be skipped until the next mode change
```
Comment (adriguir) [inline dynawo/sources/Common/Dictionaries/DYNLog_en_GB.dic:184]: I would make it more generic if we use another solver later on: initial guess of the algebraic solver is accurate

File: dynawo/sources/Simulation/DYNSimulation.cpp [hunk 2]
```diff
@@ -884,6 +886,7 @@ Simulation::updateParametersValues() {
 
 void
 Simulation::iterate() {
+  Timer timer("Simulation::iterate()");
```
Comment (adriguir) [inline dynawo/sources/Simulation/DYNSimulation.cpp:889]: I have seen in ModelMulti that they are created inside #ifdef DEBUG conditions (so only in debug compilations). As they can be printed out only in this mode, I think it is quite a good idea. What do you think?
In this case, we could apply it here too.

File: dynawo/sources/Common/Dictionaries/DYNLog_en_GB.dic (hunk 1 above)
Comment (rosiereflo) [reply dynawo/sources/Common/Dictionaries/DYNLog_en_GB.dic]: Solved

File: dynawo/sources/Simulation/DYNSimulation.cpp (hunk 2 above)
Comment (rosiereflo) [reply dynawo/sources/Simulation/DYNSimulation.cpp:897]: I totally agree.

**Pull request merged**